src/app.rs

```diff
@@ -213,7 +213,8 @@ impl<'a, T, E: Send + Sync + Clone + 'static> Application<'a, T, E> {
         #[cfg(feature = "profiler")]
         profile_scope!("initialize");
         self.states
-            .start(StateData::new(&mut self.world, &mut self.data));
+            .start(StateData::new(&mut self.world, &mut self.data))
+            .unwrap(); //TODO: figure out whether this needs to be handled
```

Comment:
`StateMachine::start` errors when there are no states available. If I understand the situation this only happens when it is run after it has previously been stopped by `stop` or `pop` until empty. Do we care about those situations here? Will a game ever be run multiple times?

Reply:
I would wrap this with a message that notifies the user that they initialized without states, I would use `expect_err` in this case.

Reply:
Wouldn't that be just plain `expect(“some message”)`? Or did I misunderstand?
```rust
self.states
    .start(start(StateData::new(&mut self.world, &mut self.data))
    .expect_err(“Tried to start state machine without any states present”);
```
Panics every time except when `start` returns an `Err`

Reply:
Yup, you are right, somehow slept read that doc 😄 . Yeah, so it should be just `expect`.